File: src/main/java/org/alham/alhamfirst/service/orchestrator/stat/TodoStatService.java
package org.alham.alhamfirst.service.orchestrator.stat;


import org.alham.alhamfirst.document.stat.StatDocument;
import org.alham.alhamfirst.dto.stat.StatDTO;

import java.util.List;
import java.util.Map;

public interface TodoStatService {


    public StatDocument saveStat(long todoIdx,  Map<String, Double> statData);

    /**
     * desc를 받아서 stat을 산출하는 서비스
     * @param desc
     */
    public StatDTO calculateStat(String desc);

    public void updateStat(String stat);

    StatDTO findByTodoIdx(long todoIdx);

    List<StatDTO> findListInTodoIdxList(List<Long> todoIdxList);
}

File: src/main/java/org/alham/alhamfirst/service/orchestrator/stat/TodoStatService.kt
package org.alham.alhamfirst.service.orchestrator.stat;


import org.alham.alhamfirst.document.stat.StatDocument;
import org.alham.alhamfirst.dto.stat.StatDTO;


interface TodoStatService {


    fun saveStat(todoIdx: Long, statData: MutableMap<String, Double>): StatDocument

    /**
     * desc를 받아서 stat을 산출하는 서비스
     * @param desc
     */
    fun calculateStat(desc: String): StatDTO
    fun updateStat(stat: String)
    fun findByTodoIdx(todoIdx: Long): StatDTO
    fun findListInTodoIdxList(todoIdxList: List<Long>): List<StatDTO>
}
